packages/security/src/CredentialManagerOverride.ts

```diff
@@ -0,0 +1,254 @@
+     */
+    public static getCredMgrInfoByPlugin(pluginName: string) : ICredentialManagerNameMap | null {
+        let credMgrInfo: ICredentialManagerNameMap = null;
+        for (const nextNameMap of this.KNOWN_CRED_MGRS) {
+            if (nextNameMap.credMgrPluginName === pluginName) {
+                credMgrInfo = nextNameMap;
+                break;
+            }
+        }
+        return credMgrInfo;
+    }
+
+    //________________________________________________________________________
+    /**
+     * Get the credential manager information for the specified plugin.
+     *
+     * @param ZEExtName - Name of the Zowe Explorer extension
+     *
+     * @returns An ICredentialManagerNameMap or
+     *          null if the specified extension is not a known credential manager.
+     */
+    public static getCredMgrInfoByZEExt(ZEExtName: string) : ICredentialManagerNameMap | null {
+        let credMgrInfo: ICredentialManagerNameMap = null;
+        for (const nextNameMap of this.KNOWN_CRED_MGRS) {
+            if (nextNameMap.credMgrZEName === ZEExtName) {
+                credMgrInfo = nextNameMap;
+                break;
+            }
+        }
+        return credMgrInfo;
+    }
+
+    //________________________________________________________________________
+    /**
+     * Get the known credential managers.
+     *
+     * @returns An array of credential managers.
+     */
+    public static getKnownCredMgrs() : ICredentialManagerNameMap[] {
+        return this.KNOWN_CRED_MGRS;
+    }
+
+    /**
+     * Override the current credential manager with the specified
+     * credential manager. A plugin or ZE extension that provides a
+     * credential manager would override the default credential manager
+     * upon installation.
+     *
+     * @param newCredMgrName
+     *        The display name of your credential manager.
+     *
+     * @throws An ImperativeError upon error.
+     */
+    public static overrideCredMgr(newCredMgrName: string) : void {
+        const credMgrInfo: ICredentialManagerNameMap =
+            CredentialManagerOverride.getCredMgrInfoByDisplayName(newCredMgrName);
+        if (credMgrInfo === null) {
+            // We do not have a known credMgr. Form a list of known credential managers.
+            const knownCredMgrs: ICredentialManagerNameMap[] = CredentialManagerOverride.getKnownCredMgrs();
+            let knownMgrsMsg: string = "";
+            for ( let credMgrInx = 0; credMgrInx < knownCredMgrs.length; credMgrInx++) {
+                knownMgrsMsg += `\n${knownCredMgrs[credMgrInx].credMgrDisplayName}`;
+            }
+
+            // we do not permit overriding by an unknown credMgr
+            throw new ImperativeError({
+                msg: `The credential manager name '${newCredMgrName}' is an unknown ` +
+                `credential manager. The previous credential manager will NOT be overridden. ` +
+                `Valid credential managers are:` + knownMgrsMsg
```

Comment:
During group design meetings, I believe that we concluded that a new credential manager "display name" will be used in settings/imperative.json to identify a credential manager instead of a plugin name. This would enable both the CLI and ZE to recognize the intended credMgr override. For the CLI to find the right plugin to call for that credMgr, it has to find the plugin name associated with the displayName. ZE must do something similar to find the right ZE extension for the credMgr. We chose to maintain a list of known credential manager overrides to enable such a lookup. You are correct that during development, a new credMgr must be placed into the known list before the new credMgr can be tested.

Are you suggesting that we should keep all of the properties (display name, plugin name, and ZE extension name) in settings/imperative.json? That seems possible, but it would likely require a plugin and a ZE Extension to provide all three names in calls to some of the CredentialManagerOverride utility functions.

Reply:
I think that much of this conversation comes from a misunderstanding of what actions these functions take. To clarify the correct behavior for future readers, I renamed the following functions.
- overrideCredMgr() to recordCredMgrInConfig()
- replaceCredMgrWithDefault to recordDefaultCredMgrInConfig